apps/confidential/confidential_test.go

```diff
@@ -1428,7 +1428,12 @@ func TestWithClaims(t *testing.T) {
 				} else if method == "credential" {
 					ar, err = client.AcquireTokenByCredential(ctx, tokenScope)
 				} else {
-					ar, err = client.acquireTokenSilentInternal(ctx, tokenScope, shared.Account{}, "", tenant, client.base.AuthParams.AuthnScheme)
+					ar, err = client.acquireTokenSilentInternal(ctx,
```

Comment:
Tests should prefer calling public methods

Reply:
Actually this is a special case, because we now have added that silent calls will return error on service principal calls .
So, we call this to just this test which tests claims

Reply:
`AcquireTokenByCredential` now does silent calls, which is only for service principal.
every other API which takes account will be able to call `AcquireTokenSilent`.
In this test the account is passed with empty object with `AcquireTokenByAuthCode` and `AcquireTokenOnBehalfOf` which fails this test, by passing this interface will have no impact on the implementation.

Reply:
let me rewrite this whole test, there are few things wrong her, It does not response with account.